build_test.go

```diff
@@ -1317,6 +1317,18 @@ func testBuild(t *testing.T, when spec.G, it spec.S) {
 				})
 
 				when("builder is untrusted", func() {
+					when("building Windows containers", func() {
+						it("errors and mentions that builder must be trusted", func() {
+							defaultBuilderImage.SetPlatform("windows", "", "")
+							h.AssertError(t, subject.Build(context.TODO(), BuildOptions{
+								Image:        "some/app",
+								Builder:      defaultBuilderName,
+								Publish:      true,
+								TrustBuilder: false,
+							}), "does not have an associated lifecycle image. Builder must be trusted.")
```

Comment:
We'll probably want to update this language to say something like "is not supported" (since there are lifecycle images for Windows now).

Reply:
Is it worth expanding the tests here: https://github.com/buildpacks/pack/blob/3a3bf282e703ee50c7f9e189015a3a9914b359a1/build_test.go#L1826 to cover more of the posix cases?

Reply:
> We'll probably want to update this language to say something like "is not supported" (since there are lifecycle images for Windows now).

This language comes from shared code, so we'd have to rework it to separate this scenario out. Given that this is temporary and the argument is somewhat specific (and Windows builds is experimental), would you be willing to allow us to punt on it?

Reply:
> Is it worth expanding the tests here:
> 
> https://github.com/buildpacks/pack/blob/3a3bf282e703ee50c7f9e189015a3a9914b359a1/build_test.go#L1826
> 
> to cover more of the posix cases?

Do you mean more `windows` cases? If so, @dfreilich asked a similar question above (see collapsed convo).

Reply:
Ah, [this convo](https://github.com/buildpacks/pack/pull/739#discussion_r467103611)? I was thinking specifically about the volume mount tests - though now that I'm reading those again, I see that they don't have to do with the OS of the container, just the host. So it's probably fine... maybe consider adding Windows-type directories here: https://github.com/buildpacks/pack/blob/3a3bf282e703ee50c7f9e189015a3a9914b359a1/build_test.go#L1801-L1808

Reply:
@natalieparellano The docker host OS and the container OS always align, so you can think of them as the same thing

Reply:
Hmm... in that case, I would expect the `windows` subset of tests to have the same number of cases as the `posix` ones, no? It looks like for Windows we just check that the drive is transformed, and on posix we test all sort of things.

Reply:
Deferred to https://github.com/buildpacks/pack/issues/793